vaadin-grid-flow-parent/vaadin-grid-flow-integration-tests/src/main/java/com/vaadin/flow/component/treegrid/it/TreeGridStringDataGenerator.java

```diff
@@ -0,0 +1,62 @@
+package com.vaadin.flow.component.treegrid.it;
+
+import com.vaadin.flow.data.provider.hierarchy.TreeData;
+
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.Optional;
+import java.util.stream.Collectors;
+import java.util.stream.IntStream;
+
+public class TreeGridStringDataGenerator {
+
+    private final TreeData<String> data = new TreeData<>();
+
+    private final Map<String, String> parentPathMap = new HashMap<>();
+
+    private Map<Integer, String> levelToNameMap;
+
+    public TreeGridStringDataGenerator() {
```

Comment:
The current API, which is based on a Map, may be difficult to understand, particularly when encountering `TreeGridStringDataGenerator` for the first time somewhere in the code. I wonder if using the builder pattern would improve it.

Example:
```java
TreeGridStringDataBuilder
  .addLevel(50)
  .addLevel(100)
  .addLevel(300)
  .build()
```
```java
TreeGridStringDataBuilder
  .addLevel(50, "Graddad")
  .addLevel(100, "Dad")
  .addLevel(300, "Son")
  .build()
```

What do you think?

Reply:
That was actually my initial approach. The reason I used the default names was the generator being used with the default names in all the cases but one. The depth was also never more than 3. This reduces the code to something like `new TreeGridStringDataGenerator().generate(3, 3, 3)`, which seemed brief and understandable. However, it is true that the builder approach is much cleaner in general, especially if it is to be used in new test pages. So, if you think that it is the better option here, I will update the PR.

Reply:
I wonder if there is a real need for custom item names in the tests. Couldn't they be always named "Item 1", "Item 1/1", etc?

Reply:
We don't really need different names since we provide the tree path anyway. However, in the [case](https://github.com/vaadin/flow-components/pull/5003/files#diff-efe17d4338c514600b81bb1ac969051a60782d3e27abc13de6e7634bdafbb3c2L79-R69) where we use the name map, it is used to provide a longer name.

Reply:
Agreed, let's leave it be 👍